### PR TITLE
feat(ui, core, localization): Add Poll attachment interactor

### PR DESCRIPTION
**Pull request by xsahil03x**
This pull request includes significant updates to the `poll` feature in the `stream_chat` package, focusing on enhancing poll functionality and improving data consistency. The changes introduce new methods for managing polls, update existing methods to use synchronization locks, and modify data models to better reflect poll states and votes.

### Enhancements to Poll Functionality:

* Added new methods for managing polls, including `deletePoll`, `closePoll`, `createPollOption`, `castPollVote`, `addPollAnswer`, `removePollVote`, and `queryPollVotes` in the `Channel` class. These methods are synchronized using `_pollLock` and `_pollVoteLock` to ensure thread safety. [[1]](diffhunk://#diff-f5b3c73311cd4ed2f91bb08aff0ca7385d4494f7e44294e082a0eb3aefee58cbR1038-R1039) [[2]](diffhunk://#diff-f5b3c73311cd4ed2f91bb08aff0ca7385d4494f7e44294e082a0eb3aefee58cbL1046-R1048) [[3]](diffhunk://#diff-f5b3c73311cd4ed2f91bb08aff0ca7385d4494f7e44294e082a0eb3aefee58cbL1059-R1163)

### Data Model Updates:

* Updated the `Poll` class to replace `answers` and `votesByOption` with `latestAnswers` and `latestVotesByOption` respectively. Added new computed properties `latestVotes`, `ownVotes`, and `ownAnswers` to provide clearer access to poll data. [[1]](diffhunk://#diff-b30c3a9ca2d3d68dec83bea192c40ceaa73d1ee4198c3804c2c25fb8f1fdf591L44-R52) [[2]](diffhunk://#diff-b30c3a9ca2d3d68dec83bea192c40ceaa73d1ee4198c3804c2c25fb8f1fdf591L112-R125) [[3]](diffhunk://#diff-b30c3a9ca2d3d68dec83bea192c40ceaa73d1ee4198c3804c2c25fb8f1fdf591R137-R152) [[4]](diffhunk://#diff-b30c3a9ca2d3d68dec83bea192c40ceaa73d1ee4198c3804c2c25fb8f1fdf591L176-R193) [[5]](diffhunk://#diff-b30c3a9ca2d3d68dec83bea192c40ceaa73d1ee4198c3804c2c25fb8f1fdf591L202-R219) [[6]](diffhunk://#diff-b30c3a9ca2d3d68dec83bea192c40ceaa73d1ee4198c3804c2c25fb8f1fdf591L254-R271)

### Synchronization Improvements:

* Introduced `_pollLock` and `_pollVoteLock` in the `Channel` class to synchronize poll-related operations, ensuring that poll actions are thread-safe. [[1]](diffhunk://#diff-f5b3c73311cd4ed2f91bb08aff0ca7385d4494f7e44294e082a0eb3aefee58cbR1038-R1039) [[2]](diffhunk://#diff-f5b3c73311cd4ed2f91bb08aff0ca7385d4494f7e44294e082a0eb3aefee58cbL1046-R1048) [[3]](diffhunk://#diff-f5b3c73311cd4ed2f91bb08aff0ca7385d4494f7e44294e082a0eb3aefee58cbL1059-R1163)

### Event Handling:

* Added new event type `pollDeleted` to the `EventType` class to handle poll deletion events.

### Testing and Documentation:

* Updated tests in `poll_test.dart` to reflect changes in the `Poll` data model, ensuring that the new properties and methods are correctly tested. [[1]](diffhunk://#diff-171a0a6ca6b2e7d1a11c426439b9c33690955595ce8a35f1e870b74a4c99a3a3L32-R32) [[2]](diffhunk://#diff-171a0a6ca6b2e7d1a11c426439b9c33690955595ce8a35f1e870b74a4c99a3a3L53) [[3]](diffhunk://#diff-171a0a6ca6b2e7d1a11c426439b9c33690955595ce8a35f1e870b74a4c99a3a3L65-R64)
* Added a new file `poll_voting_mode.dart` to define different voting modes for polls, enhancing the flexibility of poll configurations.

These changes collectively enhance the robustness and flexibility of poll management within the `stream_chat` package, ensuring better data integrity and improved functionality.

### CODE REVIEW

File: packages/stream_chat_flutter/lib/src/theme/stream_chat_theme.dart [hunk 1]
```diff
@@ -57,6 +62,10 @@ class StreamChatThemeData {
     StreamMessageListViewThemeData? messageListViewTheme,
     StreamVoiceRecordingThemeData? voiceRecordingTheme,
     StreamPollCreatorThemeData? pollCreatorTheme,
+    StreamPollInteractorThemeData? pollInteractorTheme,
+    StreamPollResultsDialogThemeData? pollResultsDialogTheme,
+    StreamPollCommentsDialogThemeData? pollCommentsDialogTheme,
+    StreamPollOptionVotesDialogThemeData? pollOptionVotesDialogTheme,
```
Comment (deven98) [inline packages/stream_chat_flutter/lib/src/theme/stream_chat_theme.dart:68]: Is the `pollOptionsDialogTheme` missing here?

**Review by sachaarbonel — DISMISSED** [review]
LGTM

File: packages/stream_chat/lib/src/core/models/poll.dart [hunk 2]
```diff
@@ -109,20 +109,20 @@ class Poll extends Equatable {
...
 
   /// List of votes received by the poll.
   ///
   /// Note: This does not include the answers provided by the users,
-  /// see [answers] for that.
-  List<PollVote> get votes => [
-        ...votesByOption.values.flattened.where((it) => !it.isAnswer),
+  /// see [latestAnswers] for that.
+  List<PollVote> get latestVotes => [
+        ...latestVotesByOption.values.flattened.where((it) => !it.isAnswer),
```
Comment (kanat) [inline packages/stream_chat/lib/src/core/models/poll.dart:120]: Does it make sense to declare this field as `final`, given that it is based on another `final` field? :)

Comment (xsahil03x) [reply packages/stream_chat/lib/src/core/models/poll.dart]: Yes, makes sense. Done

File: packages/stream_chat_flutter/lib/src/poll/stream_poll_text_field.dart [hunk 3]
```diff
@@ -3,12 +3,12 @@ import 'package:stream_chat_flutter/stream_chat_flutter.dart';
...
-/// {@template pollInputTextField}
+/// {@template streamPollTextField}
 /// A widget that represents a text field for poll input.
 /// {@endtemplate}
-class PollTextField extends StatefulWidget {
-  /// {@macro pollInputTextField}
-  const PollTextField({
+class StreamPollTextField extends StatefulWidget {
+  /// {@macro streamPollTextField}
+  const StreamPollTextField({
```
Comment (kanat) [inline packages/stream_chat_flutter/lib/src/poll/stream_poll_text_field.dart:11]: Is this a breaking change for integrators? 
- did we mention this in the changelog?
- did we apply renames to the docs?

Comment (xsahil03x) [reply packages/stream_chat_flutter/lib/src/poll/stream_poll_text_field.dart:11]: Poll isn't released yet

**Review by kanat — APPROVED** [review]

**Pull request merged**